test/addAction.test.js

```diff
@@ -0,0 +1,21 @@
+const { expect } = require('chai');
+const { Add } = require('../lib/actions/addClass');
+
+describe('Add Action', ()=> {
+    it('@asyncResolves : should return an add object containing a response', (done) => {
+        let add = new Add();
+        add.execute('label url', {links: {}})
+
+        const resolvingPromise = new Promise( (resolve) => {
+          resolve(add);
+        });
+
+        resolvingPromise.then( (result) => {
+          expect(result).to.eql({ success:true });
+          done();
+        });
```

Comment:
Based on the recommended changes to the code above, our test will change. We want to test 2 things.

1. Did our storage function act on the link data we are asking it to store? This will require something like a testing spy. Many of our Javascript devs should be able to help with that, if needed.
2. The returned promise should return some structured link data OR we should be able to catch an error, if there was a problem.

Reply:
The add some color to #2. We want the "data" that is `resolve`d by the promise to be our `Response` instance. I think we were going to name that `Response` class very similarly to the way we name our `Action` class. So if we go with `AddLinktoProjectAction` we would have `AddLinkToProjectResponse`. Just an example.

Reply:
I understand. 
I'm definitely going to have to pair on the spying.
Am I responsible for defining the AddLinkToProjectResponse for the AddLinktoProjectAction? @cromwellryan ? or should it just know to return that for now..

Reply:
I think you're going to have to at least create the class. Almost like a placeholder for the full class. We have another card (I believe) for the implementation of the full `sendViaBot` goodness.

Per my follow up about not having db setup, you can ignore the spy stuff for the moment. We don't know what that's going to look like right now.